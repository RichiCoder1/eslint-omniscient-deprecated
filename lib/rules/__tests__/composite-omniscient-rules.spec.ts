import { Linter } from "eslint";
import omniscientComponentRule from "../omniscient-component";
import RemoveOmniscientImport from "../remove-omniscient-import";
import eslintRuleConfig from "./__setup__/eslintRuleConfig";

describe("both omniscient rules", () => {
    it("should run both omniscient rule fixes", () => {
        const linter = new Linter();
        linter.defineRule("component-deprecated", omniscientComponentRule);
        linter.defineRule("no-omniscient-import", RemoveOmniscientImport);

        const report = linter.verifyAndFix(
            `import component from "omniscient";

const Test = component("Test", {
    test: 1,
    componentDidMount() {
        this.test = 2;
    }
}, () => {
    return <span>{this.test}</span>
});`,
            {
                ...eslintRuleConfig,
                rules: {
                    ["component-deprecated"]: [
                        "error",
                        {
                            areEqualModule: "ImmutableComponent",
                            areEqualImport: "isEqualImmutable",
                        },
                    ],
                    ["no-omniscient-import"]: ["error"],
                },
            }
        );

        expect(report.fixed).toBeTruthy();
        expect(report.output).toMatchSnapshot();
    });
});
