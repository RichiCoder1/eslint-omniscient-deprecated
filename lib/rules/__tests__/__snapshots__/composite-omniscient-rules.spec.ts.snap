// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`both omniscient rules should run both omniscient rule fixes 1`] = `
"import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

class Test extends Component {
    static displayName = \\"Test\\";

    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    componentDidMount = () => {
        this.test = 2;
    };

    test = 1;

    render() {
        return <span>{this.test}</span>;
    }
}"
`;
